cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1261,6 +1268,28 @@ private String getHiveDatabase(String namespace) {
     return sessionConf;
   }
 
+  /**
+   * Updates the token store to be used for the hive job, based upon the Explore container's credentials.
+   * This is because twill doesn't update the container_tokens on upon token refresh.
+   * See: https://issues.apache.org/jira/browse/TWILL-170
+   */
+  private void updateTokenStore() throws IOException, ExploreException {
+    File credentialsFile = new File(System.getenv(UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION));
+
+    File tmpFile = File.createTempFile("credentials.store", null, credentialsFile.getParentFile());
+    LOG.debug("Writing to temporary file: {}", tmpFile.getAbsoluteFile());
+
+    try (DataOutputStream os = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(tmpFile)))) {
```

Comment:
You might want to create the file with the same permission as the existing one. You can get permission and create temp file with permission using the file2 api.

http://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html